packages/spec/src/index.ts

```diff
@@ -1,6 +1,106 @@
-import type * as v1 from "./v1.ts";
+export declare namespace v1 {
+  /**
+   * The Standard Schema interface.
+   */
+  export interface StandardSchema<Input = unknown, Output = Input> {
+    /**
+     * The Standard Schema properties.
+     */
+    readonly "~standard": StandardSchemaProps<Input, Output>;
+  }
 
-export type StandardSchema = v1.StandardSchema;
+  /**
+   * The Standard Schema properties interface.
+   */
+  export interface StandardSchemaProps<Input = unknown, Output = Input> {
+    /**
+     * The version number of the standard.
+     */
+    readonly version: number;
+    /**
+     * The vendor name of the schema library.
+     */
+    readonly vendor: string;
+    /**
+     * Validates unknown input values.
+     */
+    readonly validate: (
+      value: unknown,
+    ) => StandardOutput<Output> | Promise<StandardOutput<Output>>;
+    /**
+     * Inferred types associated with the schema.
+     */
+    readonly types?: (input: Input) => Output;
+  }
 
-export type * from "./base.ts";
-export type * as v1 from "./v1.ts";
+  /**
+   * The output interface of the validate function.
+   */
+  export type StandardOutput<Output> =
+    | StandardSuccessOutput<Output>
+    | StandardFailureOutput;
+
+  /**
+   * The output interface if validation succeeds.
+   */
+  export interface StandardSuccessOutput<Output> {
+    /**
+     * The typed output value.
+     */
+    readonly value: Output;
+    /**
+     * The non-existent issues.
+     */
+    readonly issues?: undefined;
+  }
+
+  /**
+   * The output interface if validation fails.
+   */
+  export interface StandardFailureOutput {
+    /**
+     * The issues of failed validation.
+     */
+    readonly issues: ReadonlyArray<StandardIssue>;
+  }
+
+  /**
+   * The issue interface of the failure output.
+   */
+  export interface StandardIssue {
+    /**
+     * The error message of the issue.
+     */
+    readonly message: string;
+    /**
+     * The path of the issue, if any.
+     */
+    readonly path?:
+      | ReadonlyArray<PropertyKey | StandardPathSegment>
+      | undefined;
+  }
+
+  /**
+   * The path segment interface of the issue.
+   */
+  export interface StandardPathSegment {
+    /**
+     * The key representing a path segment.
+     */
+    readonly key: PropertyKey;
+  }
+
+  /**
+   * Infers the input type of a Standard Schema.
+   */
+  export type InferInput<Schema extends StandardSchema> = Parameters<
```

Comment:
This is probably DOA but I've discussed with @colinhacks that I think casing standalone types like `StandardPathSegment` as CapsCase and parameters and generics with required parameters like `inferInput` in camelCase helps visually convey more information about the purpose of a type.

That convention is used everywhere in ArkType and I think to some extent in Zod as well (e.g. `z.infer`). Maybe a bridge too far to do that in the schema as most users will be used to CapsCase everywhere, but it is a practice I think would be good for the ecosystem as types have evolved to often express more complex functionality.

Very happy to not do this, just felt obligated to throw the idea out there in case we were all aligned on it 😅

Reply:
Lol yeah too opinionated. Even though our opinion is obviously right 🙃

Reply:
I would stick to what most users are used to, but I understand why you do it differently.